### ISSUE TITLE
Split size

### ISSUE THREAD

**Issue by wojciech-kulik** [nikitabobko/AeroSpace#121]
When I open a new dialog like app settings or just another application like iOS simulator, AeroSpace splits my screen into two equal halves. 

Is it possible to configure AeroSpace so that the new window takes only as much space as it needs? Consider the iOS simulator, it is a very narrow window. Every time I open it, I have to resize Xcode.

**Comment by geert-plessers** [comment]
Same issue with Clockify Desktop, which is a narrow time tracker I like to have on the right part of my secondary monitor.

**Comment by ahmed2m** [comment]
I am facing this with Spark Mail as well. With [Amethyst](https://github.com/ianyh/Amethyst) it was enforcing the visible portion of the window rules according to the layout but only expanding the said window over other windows if it is focused and has ANY visible part off the screen. It was an elegant solution tbh, would love to see it, if it's an easy fix to implement in Aerospace.

**Comment by ahmed2m** [comment]
Showcasing the issue for the minimum size of the window for Spark mail:

https://github.com/nikitabobko/AeroSpace/assets/26524089/c05738a1-8c56-40b6-b2cd-0d149758b04a

**Comment by jbrodriguez** [comment]
i have this same issue with ios simulator or android emulator, i have a cpl of ideas
- creating a small app to show a transparent and click-through window with a specific width so that when simulator/emulator are opened we could open this app to reserve space (far fetched)
- have per-window padding ? so for ios we'd have -100 horizontal padding (an example), for android -50 (android window is bigger), so that the window on the other half would get strechted

**Comment by jbrodriguez** [comment]
i was thinking of a third option
- per window min width

not sure how nice these options play with the split algorithm

**Comment by KingSupernova31** [comment]
Agreed this is a big problem; when adding a new window to a container, Aerospace will often try to give it a size that conflicts with its minimum size, resulting in the window going offscreen or overlapping other windows.

And any attempt to resize windows when near their minimum size often causes chaos with the layout, causing other windows to resize unnecessarily too, or get stuck with empty space in between, or change to a different size after I release the mouse.
